### DISCUSSION TITLE
synced-lyrics: romanization

### DISCUSSION THREAD

**Discussion by ArjixWasTaken** [pear-devs/pear-desktop#2382] [General]
I plan to add romanization support to the plugin `synced-lyrics`, for that we'll need the addition of the following libraries:

1. https://github.com/hotoo/pinyin (chinese)
2. https://github.com/toss/es-hangul (korean)
3. https://github.com/hexenq/kuroshiro (japanese)

I am making this discussion so we can discuss if the above list should be replaced with better libraries, or more libraries to be added for more languages.

cc: @JellyBrick, @Su-Yong

**Comment by Su-Yong** [comment]
I think [toss/es-hangul](https://github.com/toss/es-hangul) is the best choice for Korean.
It is maintained by toss that is big tech company in Korean. So it seems to be a more reliable than [fujaru/aromanize-js](https://github.com/fujaru/aromanize-js)

here is [a doc](https://es-hangul.slash.page/en/docs/api/romanize) about romanize

**Comment by JellyBrick** [comment]
For Korean, I recommend this library (made by a famous bank in Korea): https://www.npmjs.com/package/es-hangul

see https://es-hangul.slash.page/en/docs/api/romanize.en

**Comment by JellyBrick** [comment]
And, for Japanese, I recommend https://github.com/hexenq/kuroshiro

**Comment by Su-Yong** [comment]
Japanese has some issue. I'd want to use this [hexenq/kuroshio](https://github.com/hexenq/kuroshiro). But in description, it doesn't support romanize function because converting furigana to romaji is not matched real pronunciation.
> For example, you'll get "kousi", "koushi", "koushi" respectively when converts kana "こうし" to romaji using nippon, passport, hepburn romanization system.

for more details, see https://green.adam.ne.jp/roomazi/onamae.html#naze

So, We will use kuroshiro, We need to map kana to romaji

**Comment by okinawaa** [comment]
Thank you to everyone who has mentioned es-hangul.
I’m the developer managing this project. If es-hangul doesn’t provide a feature you need, please feel free to reach out at any time.
I’ll be happy to support you promptly and to the highest standard.

**Reply by JellyBrick** [reply]
`es-hangul` has now been applied to the project! Thanks!

https://github.com/th-ch/youtube-music/commit/4b35a967782e2e435d7942d814825545d104d33e

**Comment by ArjixWasTaken** [comment]
Made a PR, [here](https://github.com/th-ch/youtube-music/pull/2790)!

**Comment by AARP41298** [comment]
Hello, just a personal fun fact:
For Korean letters, there is the McCune romanization system. I've been listening to K-pop for years, and since I discovered this method a few months ago, singing the songs has become much easier.

The best website I've worked with is https://www.laotouzi.de/conv/ (specifically the second box that uses https://www.laotouzi.de/conv/neuer_versuch.js).

Because of how it works, some English phrases gets translated (and broken), and some characters seem to get lost (??).

**Comment by Andi9910** [comment]
im just going back here and wants to thank you all ❤️
